internal/buildinfo/version.go

```diff
@@ -2,23 +2,164 @@ package buildinfo
 
 import (
 	"errors"
-	"fmt"
-	"os"
 	"runtime/debug"
-	"strconv"
+	"strings"
 	"time"
 
 	"github.com/blang/semver"
+	"github.com/loadsmart/calver-go/calver"
 )
 
+type Version interface {
+	String() string
+	EQ(other Version) bool
+	Newer() bool
+	SeverelyOutdated(other Version) bool
+}
+
+type SemverVersion struct {
+	semver.Version
+}
+
+func ParseSemver(version string) (*SemverVersion, error) {
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, err
+	}
+	return &SemverVersion{v}, nil
+}
+
+func (v SemverVersion) EQ(other Version) bool {
+	o, ok := other.(*SemverVersion)
+	if ok {
+		return v.Version.EQ(o.Version)
+	} else {
+		return false
+	}
+}
+
+func (v *SemverVersion) Newer() bool {
+	_, ok := parsedVersion.(*CalverVersion)
+	if ok {
+		return false
+	} else {
+		other := parsedVersion.(*SemverVersion)
+		return v.Version.GT(other.Version)
+	}
+}
+
+func (v *SemverVersion) SeverelyOutdated(other Version) bool {
+	otherVer, ok := other.(*SemverVersion)
+	if ok {
+		if v.Major < otherVer.Major {
+			return true
+		}
+		if v.Minor < otherVer.Minor {
+			return true
+		}
+		if v.Patch+5 < otherVer.Patch {
+			return true
+		}
+		return false
+	} else {
+		return false
+	}
+}
+
+type CalverVersion struct {
+	calver.Version
+}
+
+const calverFormat = "YYYY.0M.0D.MICRO"
+
+func ParseCalver(version string) (*CalverVersion, error) {
+	v, err := calver.Parse(calverFormat, version)
+	if err != nil {
+		// Does the version parse as calver without dashes? Needed for semverr
+		// compatibility for goreleaser.
+		dedashedVersion := strings.Replace(version, "-", ".", 1)
+		v, err = calver.Parse(calverFormat, dedashedVersion)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &CalverVersion{*v}, nil
+}
+
+func (v CalverVersion) EQ(other Version) bool {
+	o, ok := other.(*CalverVersion)
+	if ok {
+		return v.Version.CompareTo(&o.Version) == 0
+	} else {
+		return false
+	}
+}
+
+func (v *CalverVersion) Newer() bool {
+	other, ok := parsedVersion.(*CalverVersion)
+	if ok {
+		return v.Version.CompareTo(&other.Version) == 1
+	} else {
+		return true
+	}
+}
+
+func (v *CalverVersion) SeverelyOutdated(other Version) bool {
+	// todo: Figure out how to do this when we actually have outdated calver
+	// versions.
+	return false
```

Comment:
I really wish calver-go had an API for getting a time.Time from a version. That's really what this needs...

Reply:
[My fork](https://github.com/ndarilek/calver-go) does. I've PR'd the changes--if they aren't merged upstream, I'm happy to move that over to superfly.

Reply:
In that case, why don't we just start with checking if the version is a month or two old? We can always adjust if needed later :)

Reply:
Sure thing. Is there a more idiomatic way of doing date comparisons in golang other than checking if the timestamps are > something like 86400 * 30 seconds apart? That was my initial inclination but figured I'd save a round trip if there's a better way.

Reply:
I think that's a perfectly fine way to implement that :)